sklearn/semi_supervised/label_propagation.py

```diff
@@ -264,12 +259,15 @@ def fit(self, X, y):
 
         l_previous = np.zeros((self.X_.shape[0], n_classes))
 
-        remaining_iter = self.max_iter
         unlabeled = unlabeled[:, np.newaxis]
         if sparse.isspmatrix(graph_matrix):
             graph_matrix = graph_matrix.tocsr()
-        while (_not_converged(self.label_distributions_, l_previous, self.tol)
-               and remaining_iter > 1):
+
+        self.n_iter_ = 0
+        while self.n_iter_ < self.max_iter:
```

Comment:
You mean setting `self.n_iter_` using the loop variable?
I personally feel a bit uncomfortable using the loop variable outside the loop. 

(* angsty feeling *)